File: src/api-dotnet/Prs.Bootcamp/Data/Models/PurchaseRequestLineItem.cs
using System.ComponentModel.DataAnnotations;

namespace Prs.Bootcamp.Data.Models;

public class PurchaseRequestLineItem
{
    [Required]
    public int Id { get; set; }

    [Required]
    public int PurchaseRequestId { get; set; }
    public PurchaseRequest? PurchaseRequestNavigation { get; set; }

    [Required]
    public int ProductId { get; set; }
    public Product? ProductNavigation { get; set; }

    [Required]
    public int Quantity { get; set; }

    [Required]
    public bool IsFulfilled { get; set; }

    public void Copy(PurchaseRequestLineItem lineItem)
    {
        Id = lineItem.Id;
        PurchaseRequestId = lineItem.PurchaseRequestId;
        ProductId = lineItem.ProductId;
        Quantity = lineItem.Quantity;
        IsFulfilled = lineItem.IsFulfilled;
    }
}

File: src/api-java/src/main/java/app/bootcamp/prs/model/PurchaseRequestLineItem.java
package app.bootcamp.prs.model;

import jakarta.persistence.*;

@Entity
@Table(name = "purchase_request_line_item")
public class PurchaseRequestLineItem {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @ManyToOne
    @JoinColumn(name = "purchase_request_id", nullable = false)
    private PurchaseRequest purchaseRequest;

    @ManyToOne
    @JoinColumn(name = "product_id", nullable = false)
    private Product product;

    @Column(name = "quantity", nullable = false)
    private Integer quantity;

    // Getters and Setters
    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public PurchaseRequest getPurchaseRequest() {
        return purchaseRequest;
    }

    public void setPurchaseRequest(PurchaseRequest purchaseRequest) {
        this.purchaseRequest = purchaseRequest;
    }

    public Product getProduct() {
        return product;
    }

    public void setProduct(Product product) {
        this.product = product;
    }

    public Integer getQuantity() {
        return quantity;
    }

    public void setQuantity(Integer quantity) {
        this.quantity = quantity;
    }
}